jenkins/client/init.sls

```diff
@@ -3,6 +3,7 @@
 
 include:
   - jenkins.client.plugin
+  - jenkins.client.security
```

Comment:
You should wrap into condition like `if client.security is defined`, the same probably also for credentials, jobs, etc.